licensing/src/main/java/io/helidon/build/licensing/Report.java

```diff
@@ -330,10 +363,169 @@ private boolean generateAttributionFile(AttributionDocument attributionDocument,
+                    }
+                }
+                w.write("\t\t\t]\n");
+                w.write("\t\t}\n");
+
+                detectLicenses(licensesUsed, d.getAttribution());
+            }
+        }
+        w.write("\t]\n");
+
+        // If we haven't written anything, then let the caller know
+        if (first) {
+            w.write("}\n");
+            return false;
+        }
+
+        // Write full text of licenses used (that were squashed out of report) to the
+        // end of the file.
+        first = true;
+        for (String s : licensesUsed) {
+            if (first) {
+                w.write("\t, \"licenses\" : {\n");
+            }
+
+            // Get license text for AttributionDocument
+            AttributionLicense license = getLicense(attributionDocument, s);
+            String sJson = jsonEscape(s);
+            if (license != null) {
+                w.write(String.format("\t\t%s\"%s\": \"%s\"\n", (first ? "":", "), sJson, jsonEscape(license.getText())));
+            } else {
+                w.write(String.format("\t\t%s\"%s\": \"\"No license text found for %s\"\n", (first ? "":", "), sJson, sJson));
+            }
+
+            first = false;
+        }
+        w.write("\t}\n");
+
+        w.write("}\n");
+
+        return true;
+    }
+
+    private static String jsonEscape(String str) {
+        return str
+                .replaceAll("\\\\", "\\\\\\\\")
+                .replaceAll("\n", "\\\\n")
+                .replaceAll("\"", "\\\\\"");
+    }
+
+    /**
+     * Generates a third party attribution file from all found BAs in XLS format.
+     *
+     * @param attributionDocument AttributionDocument to generate attribution report from
+     * @param w                   FileWriter to write report to
+     * @return true if something was written, else false
+     * @throws IOException if trouble writing to file
+     */
+    private boolean generateAttributionFileHtml(AttributionDocument attributionDocument, FileWriter w) throws IOException {
+        w.write("<html>\n");
+        w.write("<body>\n");
+        w.write("<h1>Third Party Attributions");
+        w.write("<table border=1>");
+        w.write(String.format("<tr><th>Name</th>%s<th>Licensor</th><th>License Name</th>", includeVersion ? "<th>Version</th>":""));
+        w.write("<th>Attribution</th><th>Used By</th></tr>\n");
+        List<AttributionDependency> deps = attributionDocument.getDependencies();
+        Set<String> licensesUsed = new HashSet<>();
+        boolean first = true;
+        for (AttributionDependency d : deps) {
+            HashSet<String> intersection = new HashSet<>(moduleList);
+            intersection.retainAll(d.getConsumers());
+            if (moduleList.isEmpty() || !intersection.isEmpty()) {
+                w.write("<tr valign=top>");
+                w.write(String.format("<td>%s</td>", d.getName()));
+                if (includeVersion)
```

Comment:
Please run `mvn validate  -Pcheckstyle` and fix the few, minor code style errors.